### PR TITLE
Issue-1438: Fix language-related error for testJWTAuthShouldFailWithoutToken test.

### LINKED ISSUE #1438: Test fails due to JwtAuthenticationTest
# Bug Description
Testing JwtAuthenticationTest.testJWTAuthShouldFailWithoutToken (which automatically occurs when running `mvn clean install`) on Windows 10 Pro fails with the following error:
```
JwtAuthenticationTest.testJWTAuthShouldFailWithoutToken:99 1 expectation failed.
JSON path error doesn't match.
Expected: Full authentication is required to access this resource
  Actual: Vollstõndige Authentifikation wird ben÷tigt um auf diese Resource zuzugreifen
```
Changing the system language to english fixes the error.

# Steps To Reproduce

1. Change system language to german
2. clone the repo
3. `cd strongbox`
4. `mvn test -Dtest=JwtAuthenticationTest#testJWTAuthShouldFailWithoutToken -pl strongbox-web-core`

# Expected Behavior
The test should not be dependant on system language and pass on other languages than english. 

# Environment

* Strongbox Version: 0d306ed374652714cd96a40b012e9df30b4f40e1
* OS:  Windows 10 Pro
* Maven version:
```
Apache Maven 3.5.0 (ff8f5e7444045639af65f6095c62210b5713f426; 2017-04-03T21:39:06+02:00)
Maven home: C:\apache-maven-3.5.0
Java version: 1.8.0_221, vendor: Oracle Corporation
Java home: C:\Program Files\Java\jdk1.8.0_221\jre
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
``` 

# Additional context
The error message is hardcoded into the test as the String variable UNAUTHORIZED_MESSAGE. This causes the test to fail if the error message isn't the original english error message. 

# Task Relationships

This bug:
* Is caused by: 
* Relates to: 
* Depends on:

### PR DESCRIPTION
**Pull request by ptirador**
# Pull Request Description

This pull request closes #1438 

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

**Comment by carlspring** [comment]
Thanks, @ptirador ! I was thinking of the same sort of fix!

@zernerus: Would you mind trying this out and reporting back? Thanks!

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/JwtAuthenticationTest.java [hunk 1]
```diff
@@ -89,14 +95,20 @@ public void testJWTAuthShouldPassWithToken()
     @Test
     public void testJWTAuthShouldFailWithoutToken()
     {
+        String defaultExpectedErrorMessage = messages.getMessage(UNAUTHORIZED_MESSAGE_CODE,
+                                                                 Locale.ENGLISH);
+        String expectedErrorMessage = messages.getMessage(UNAUTHORIZED_MESSAGE_CODE,
+                                                          defaultExpectedErrorMessage);
+        String utf8ExpectedErrorMessage = new String(expectedErrorMessage.getBytes(ISO_8859_1), UTF_8);
```
Comment (Zernerus) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/JwtAuthenticationTest.java]:
```suggestion
        String utf8ExpectedErrorMessage = new String(expectedErrorMessage.getBytes(ISO_8859_1), Charset.defaultCharset());
```

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/JwtAuthenticationTest.java [hunk 2]
```diff
@@ -3,45 +3,51 @@
 import org.carlspring.strongbox.config.IntegrationTest;
 import org.carlspring.strongbox.rest.common.RestAssuredBaseTest;
 import org.carlspring.strongbox.users.security.SecurityTokenProvider;
+
+import javax.inject.Inject;
```
Comment (Zernerus) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/JwtAuthenticationTest.java]:
```suggestion
import javax.inject.Inject;
import java.nio.charset.Charset;
```

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/JwtAuthenticationTest.java [hunk 3]
```diff
@@ -3,45 +3,51 @@
...
+import org.springframework.security.core.SpringSecurityMessageSource;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.test.context.TestSecurityContextHolder;
-
-import javax.inject.Inject;
-import java.util.Collections;
-
 import static io.restassured.module.mockmvc.RestAssuredMockMvc.given;
+import static java.nio.charset.StandardCharsets.ISO_8859_1;
+import static java.nio.charset.StandardCharsets.UTF_8;
```
Comment (Zernerus) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/JwtAuthenticationTest.java]:
```suggestion

```

**Comment by ptirador** [comment]
> The default Charset of Windows is windows-1252, so using UTF-8 doesn't work here.
> What worked for me is importing **java.nio.charset.Charset** and using the **defaultCharset** method to determine the right charset.
> It works for me on windows but I need someone to test it on other operation systems aswell.

Thanks so much @Zernerus! I've just tried on Ubuntu and it works fine.

**Comment by carlspring** [comment]
Thanks, @ptirador for fixing this and thanks, @Zernerus for reporting and reviewing it! :)

**Pull request merged**